core/actions/startDeploy.js

```diff
@@ -17,6 +17,9 @@ module.exports = (
     try {
       deploysController.start();
     } catch (e) {
+      notify(repos, 'deployInProgress', verbose, err => {
```

Comment:
`return cb(e);` should be called inside this callback.
And the `err` from notify isn't about the deploy in progress it could be anything related to the slack integration, so just leave it as
`if (err) logger.error("Error notifying slack", repos, err);`

In summary:
```javascript
notify(repos, 'deployInProgress', verbose, err => {
  if (err) logger.error("Error notifying slack", repos, err);
  cb(e);
});
```